src/app/modules/main/communities/minting/models/token_item.nim

```diff
@@ -0,0 +1,100 @@
+import strformat
+
+type
+  TokenType* {.pure.} = enum
+    #ERC20,
+    ERC721
+
+type
+  MintingState* {.pure.} = enum
+    InProgress,
+    Minted,
+    Failed
+
+type
+  TokenItem* = ref object
+    tokenType: TokenType
+    tokenAddress: string
+    name: string
+    description: string
+    icon: string
+    supply: int
+    infiniteSupply: bool
+    transferable: bool
+    remoteSelfDestruct: bool
+    networkId: string
+    mintingState: MintingState
+    # TODO holders
+
+proc initCollectibleTokenItem*(
+  tokenAddress: string,
+  name: string,
+  description: string,
+  icon: string,
+  supply: int,
+  infiniteSupply: bool,
+  transferable: bool,
+  remoteSelfDestruct: bool,
+  networkId: string,
+  mintingState: MintingState
+): TokenItem =
+  result = TokenItem()
+  result.tokenType = TokenType.ERC721
+  result.tokenAddress = tokenAddress
+  result.name = name
+  result.description = description
+  result.icon = icon
+  result.supply = supply
+  result.infiniteSupply = infiniteSupply
+  result.transferable  = transferable
+  result.remoteSelfDestruct = remoteSelfDestruct
+  result.networkId = networkId
+  result.mintingState = mintingState
+
+proc `$`*(self: TokenItem): string =
+  result = fmt"""TokenItem(
+    tokenType: {$self.tokenType.int},
+    tokenAddress: {self.tokenAddress},
+    name: {self.name},
+    description: {self.description},
+    icon: {self.icon},
+    supply: {self.supply},
+    infiniteSupply: {self.infiniteSupply},
+    transferable: {self.transferable},
+    remoteSelfDestruct: {self.remoteSelfDestruct},
+    networkId: {self.networkId},
+    mintingState: {$self.mintingState.int}
+    ]"""
+
+proc getTokenType*(self: TokenItem): TokenType =
+  return self.tokenType
+
+proc getTokenAddress*(self: TokenItem): string =
+  return self.tokenAddress
+
+proc getName*(self: TokenItem): string =
+  return self.name
+
+proc getDescription*(self: TokenItem): string =
+  return self.description
+
+proc getIcon*(self: TokenItem): string =
+  return self.icon
+
+proc getSupply*(self: TokenItem): int =
+  return self.supply
```

Comment:
Any chance of this being a very big number? If so, we should pass it as a string and use some BIgInt lib

Reply:
I had string here. However you can see: https://github.com/status-im/status-desktop/pull/9082#issuecomment-1403806397. Because our limit is 999999999 we can use int for now.